MonoGame.Framework/FrameworkDispatcher.cs

```diff
@@ -2,15 +2,57 @@
 // This file is subject to the terms and conditions defined in
 // file 'LICENSE.txt', which is part of this source code package.
 
+using System;
+using Microsoft.Xna.Framework.Audio;
+
 namespace Microsoft.Xna.Framework
 {
-	public static class FrameworkDispatcher
-	{
-		public static void Update()
-		{
-			// "Updates the status of various framework components
-			// (such as power state and media), and raises related events"
-		}
-	}
+    /// <summary>
+    /// Helper class for processing internal framework events.
+    /// </summary>
+    /// <remarks>
+    /// If you use <see cref="Game"/> class, <see cref="Update()"/> is called automatically.
+    /// Otherwise you must call it as part of your game loop.
+    /// </remarks>
+    public static class FrameworkDispatcher
+    {
+        private static bool _initialized = false;
+
+        /// <summary>
+        /// Processes framework events.
+        /// </summary>
+        public static void Update()
+        {
+            if (!_initialized)
+                Initialize();
+
+            DoUpdate();
+        }
+
+        private static void DoUpdate()
+        {
+            DynamicSoundEffectInstanceManager.UpdatePlayingInstances();
+            SoundEffectInstancePool.Update();
+        }
+
+        private static void Initialize()
+        {
+            // Initialize sound system
+#if DIRECTX
+            SoundEffect.InitializeSoundEffect();
+#elif OPENGL
+            try
+            {
+                var oal = OpenALSoundController.GetInstance;
+            }
+            catch (DllNotFoundException ex)
+            {
+                throw new NoAudioHardwareException("Failed to init OpenALSoundController", ex);
+            }
+#endif

```

Comment:
Now that you mention it, this is indeed a breaking change. Previously, the audio device was created in `SoundEffect` static constructor (which could be called on any thread, see #1494). Now, a call to `FrameworkDispatcher.Update()` is required. On the other hand, just about all MSDN docs for XNA say things like

> Play throws an InvalidOperationException if FrameworkDispatcher.Update has not been called at least once before making this call.

The line applies to Silverlight apps, but it was just as true when I was working on `DynamicSoundEffectInstance` tests.

I guess there could be some logic in sound effect initialization to allow both cases, if we want to preserve the old behavior.